### PR TITLE
fix: 209 Calling `new Array` with 1 argument doesn't work properly

### PR DESCRIPTION
**Pull request by cisen**
#209

### CODE REVIEW

File: src/lib/builtins/array.rs [hunk 1]
```diff
@@ -85,11 +85,19 @@ pub fn make_array(this: &Value, args: &[Value], ctx: &mut Interpreter) -> Result
     // between indices and values): this creates an Object with no prototype
 
     // Create length
-    let length = Property::new()
-        .value(to_value(args.len() as i32))
-        .writable(true)
-        .configurable(false)
-        .enumerable(false);
+    let length = if args.len() > 1 {
```
Comment (IovoslavIovchev) [inline src/lib/builtins/array.rs]: You can extract just the length and write the logic for creating the property once, e.g.
```rust
let n = if args.len() > 1 { args.len() } ...
```

**Comment by cisen** [comment]
> LGTM! Maybe a `match` statement would be a better fit instead of the `if` statements and would also allow you to recapture the value inside the pattern match.
> Not a big change but would make it easier to read and express intent better IMO.

Thanks for your advice! I will improve it later. Because of Chinese New Year and my hometown's slow network，It will take a little time.

**Comment by IovoslavIovchev** [comment]
Looks good but can you look for some commented tests for this or add some if they don't exist?

**Comment by cisen** [comment]
> Looks good but can you look for some commented tests for this or add some if they don't exist?

Ok, I will add it later.

**Comment by cisen** [comment]
There is still some problems when I add tests like [jerryscripts'](https://github.com/jerryscript-project/jerryscript/blob/master/tests/jerry/array.js). It will will take more times to fix this problems.

**Pull request closed without merging**